service/controller/app/resource/chart/update.go

```diff
@@ -60,24 +66,31 @@ func (r *Resource) newUpdateChange(ctx context.Context, currentResource, desired
 		return nil, microerror.Mask(err)
 	}
 
+	if reflect.DeepEqual(currentChart, &v1alpha1.Chart{}) {
+		return &v1alpha1.App{}, nil
+	}
+
 	desiredChart, err := toChart(desiredResource)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.Debugf(ctx, "finding out if the chart has to be updated")
-
 	updateChart := &v1alpha1.Chart{}
-	isModified := !isEmpty(currentChart) && !equals(currentChart, desiredChart)
-	if isModified {
-		r.logger.Debugf(ctx, "the chart has to be updated")
+
+	resourceVersion := currentChart.GetResourceVersion()
+
+	currentChart = copyChart(currentChart)
+	copyAnnotation(currentChart, desiredChart)
```

Comment:
Can you add comments explaining what copyChart and copyAnnotations do? So its clearer why we need to do this.